python/osrd_schemas/osrd_schemas/rolling_stock.py

```diff
@@ -169,6 +170,26 @@ class EtcsBrakeParams(BaseModel, extra="forbid"):
     t_be: float = Field(ge=0, description="T_be: safe brake build up time in s")
 
 
+class SimpleSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"]
+
+    @model_serializer
+    def ser(self) -> Mapping[str, Any]:
+        return {"type": self.signaling_system_type}
+
+
+class EtcsSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["ETCS_LEVEL2"]
+    brake_params: EtcsBrakeParams
+
+    @model_serializer
+    def ser(self) -> Mapping[str, Any]:
+        return {
+            "type": self.signaling_system_type,
+            "brake_params": self.brake_params.model_dump(),
+        }
```

Comment:
I think that you can achieve similar results with only
```suggestion
class SimpleSignalingSystem(BaseModel, extra="forbid"):
    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"] = Field(alias="type")


class EtcsSignalingSystem(BaseModel, extra="forbid"):
    signaling_system_type: Literal["ETCS_LEVEL2"] = Field(alias="type")
    brake_params: EtcsBrakeParams
```

Reply:
OK, after more testing locally, here are the options we have that work on the client side by order of preference:
1. just use `type` as param name (seems to work and pass CI for client, I don't know what was the issue initially 😬)
2. use alias, bump `pydantic` to `>= 2.11` (so `pydantic-core ~= 2.33.2`) to have [serialize_by_alias](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.serialize_by_alias) available (ditto for ETCS):
```python
class SimpleSignalingSystem(BaseModel, extra="forbid"):
    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"] = Field(alias="type")

    model_config = ConfigDict(serialize_by_alias=True)
```
3. use alias, and tell every call of `model_dump` (and any other serializer) to use alias via model_dump(`by_alias=True`) 🤯 (that's what broke our initial attempt)

Reply:
I have a slight preference in favor of option 2.

You won't actually shadow the builtin `type` function with 1., but I don't think it's a best practice in general to name anything like builtins

Reply:
done in https://github.com/OpenRailAssociation/osrd/pull/13435/commits/4d0efb625492f081542d410931521701bc5cddc4, I chose option 1 because I tested it and it looks like it works on my machine.

Reply:
@Wadjetz the point wasn't about whether it would work or not.

Wrong commit linked

Reply:
Little preference for 2️⃣  after your argument @Castavo, then.

Option 1️⃣ is propagated in client ✔️

In the client I bumped pydantic to also support option 2️⃣ (the same should be done in osrd-schemas)
<img width="686" height="156" alt="image" src="https://github.com/user-attachments/assets/f38998cd-e527-4f40-a76f-7bce02f5ed0d" />

Reply:
done in https://github.com/OpenRailAssociation/osrd/pull/13435/commits/d21c253248168cdc0c745caf988163a246ba727c

Reply:
OK, it works on client's side now.

It's not entirely settled there, as `ty` has to be bumped for that and messes with the CI.